visits-service-new/src/main/java/com/petclinic/visits/visitsservicenew/BusinessLayer/VisitServiceImpl.java

```diff
@@ -83,10 +84,25 @@ public Mono<VisitResponseDTO> addVisit(Mono<VisitRequestDTO> visitRequestDTOMono
                 .map(EntityDtoUtil::toVisitEntity)
                 .doOnNext(x -> x.setVisitId(EntityDtoUtil.generateVisitIdString()))
                 .doOnNext(v -> System.out.println("Entity Date: " + v.getVisitDate())) // Debugging
-                .flatMap((repo::insert))
-                .map(EntityDtoUtil::toVisitResponseDTO);
+                .flatMap(visit ->
```

Comment:
[Nitpick]: this verification should occur before generating an id for the visit to prevent unnecessary operations from being performed if a request is invalid.